subxt/src/client.rs

```diff
@@ -354,6 +354,18 @@ where
             .await
     }
 
+    /// Return the SCALE encoded bytes representing the call data of the transaction.
+    pub fn call_data(&self) -> Result<Vec<u8>, BasicError> {
+        let mut bytes = Vec::new();
+        let locked_metadata = self.client.metadata();
+        let metadata = locked_metadata.read();
```

Comment:
This pattern is very common to use the read only locked metadata, I wonder whether it is worth adding a method for that.

Reply:
Mmm yeah, quite possibly; it might also be worth just cloning it out of the lock immediately too; no need to actually hold the lock for any appreciable amount of time either I think!

Reply:
Cloning the metadata seems a bit wasteful in most cases since as you say it is usually short lived